src/middlewared/middlewared/etc_files/pam_tdb.py

```diff
@@ -0,0 +1,43 @@
+from middlewared.utils.user_api_key import (
+    UserApiKey,
+    PamTdbEntry,
+    flush_user_api_keys
+)
+
+
+def convert_keys(username, keys) -> PamTdbEntry:
+    user_api_keys = []
+
+    for key in keys:
+        if key['expires_at'] is None:
+            expiry = 0
+        else:
```

Comment:
We actually filter out `revoked` when we `etc.generate`. That said I'll add error handling here in case someone helpfully edits `plugins/etc.py` without understanding context.